src/monattack.cpp

```diff
@@ -1329,9 +1329,12 @@ static body_part body_part_hit_by_plant()
 bool mattack::growplants( monster *z )
 {
     for( const auto &p : g->m.points_in_radius( z->pos(), 3 ) ) {
-        // TODO: Make this sensible - it can destroy EVERYTHING
-        if( !g->m.has_flag( "DIGGABLE", p ) && one_in( 4 ) ) {
-            g->m.ter_set( p, t_dirt );
+
+        // Only affect natural, dirtlike terrain, or terrain of the
+        // types we create (it's important that we can destroy our own
+        // trees, so we can't trap ourself).
+        if( !( g->m.has_flag_ter( "DIGGABLE", p ) ||
+               g->m.ter( p ) == t_tree_young || g->m.ter( p ) == t_tree ) ) {
```

Comment:
I recommend checking for the "TREE" flag instead of "t_tree". There are all kinds of trees in the game, and they all have different ids.

Reply:
I know, but I didn't think we would want to allow triffid queens to destroy fruit trees, etc.; the only reason to allow tree destruction at all is to keep them from getting trapped.

Reply:
Outworld triffids won't differentiate between Earth plant genera.

Reply:
Okay, done. We still need an explicit check `t_tree_young` because it's not technically a `TREE`.